rskj-core/src/main/java/co/rsk/util/EthSwapUtil.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.util;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.core.bc.ClaimTransactionInfoHolder;
+import co.rsk.db.RepositorySnapshot;
+import org.ethereum.config.Constants;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.solidity.SolidityType;
+import org.ethereum.util.ByteUtil;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.GasCost;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+public class EthSwapUtil {
```

Comment:
Hey guys I'm not sure about this utility class. There is only 4 methods called from other classes and we have 7 public methods. 

When methods are moved to utility clases, the reason usually is because they are going to be used in different contexts and there is no need to instantiate a new object for doing this kind of operations.

But in this case it seems that there is only one specific usage related with specific kind of Transaction. 

These kind of classes could be dangerous because it could be easy to lost the context of it when working with then. Also the own class or some of the  methods could grow or being modified ending in situations hard to maintain or leading to errors.

Reply:
Fair point. I can move this logic to a instantiable class, maybe combine this with the ClaimTransactionInfoHolder (changing the name of course) since I am creating an instance for that anyways. Wdyt? @asoto-iov @jurajpiar @fmacleal.

Reply:
Have you think about create a new type of Transaction?
As an example we have the RemascTransaction maybe we can have a ClaimTransaction which extends Transaction to contain the logic related to it.

Reply:
We can discuss this idea with the team.